src/Core/Managed/Net40/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -44,8 +44,27 @@ protected TelemetryConfigurationFactory()
         /// </remarks>
         public static TelemetryConfigurationFactory Instance
         {
-            get { return instance ?? (instance = new TelemetryConfigurationFactory()); }
-            set { instance = value; }
+            // Both get and set should write something to CoreEventSource so as to initialize it. This is a fix for below deadlock issue   
```

Comment:
Yes if default config is used. If user passed an instance of IChannel to TelemetryConfiguration, then the deadlock can still occur. But attempt to use the default TelemetryConfiguration will internally use TelemetryConfigurationFactory.Instance.